### PR TITLE
load /// comments for System.Buffers APIs

### LINKED ISSUE #1730: IBufferWriter<T> documentation is missing
There is no API documentation - https://docs.microsoft.com/en-us/dotnet/api/system.buffers.ibufferwriter-1

Many other APIs in System.Buffers are missing API documentation - https://docs.microsoft.com/en-us/dotnet/api/system.buffers

### PR DESCRIPTION
**Pull request by mairaw**
Made some minor edits but it's mostly straight from source

Fixes #1730 
Contributes to #1010

### CODE REVIEW

**Comment by rpetrusha** [comment]
The build failed because "an item with the same key has already been added" in the Base64 type. However, there don't appear to be any duplicate entries, and this PR doesn't change any of the basic data gathered through reflection. Closing and reopening in the hope that a new build will succeed.

**Comment by mairaw** [comment]
@TianqiZhang can you help us diagnose this error? Thx!

**Comment by mairaw** [comment]
This is what auto-generated documentation looks like 😆

**Comment by mairaw** [comment]
Thanks so much for your thorough review @rpetrusha. Next time, don't worry so much the missing sections when we're auto-loading comments, otherwise, it will take too much of your time reviewing but I really appreciate it ❤️

**Comment by rpetrusha** [comment]
Thanks, @mairaw. I've never been fond of auto-generated documentation, and I'm sure I went overboard. But you've done a real service in getting this documented by importing triple-slash comments.

**Comment by ahsonkhan** [comment]
Do we have plans to review the contents and update the docs to be more readable/user friendly? I suspect the /// comments from the source might not be detailed enough.

File: xml/System.Buffers/IBufferWriter`1.xml [hunk 1]
```diff
@@ -15,8 +15,8 @@
   </TypeParameters>
   <Interfaces />
   <Docs>
-    <typeparam name="T">To be added.</typeparam>
-    <summary>To be added.</summary>
+    <typeparam name="T">The type of the items in the <see cref="T:System.Buffers.ReadOnlySequence`1" />.</typeparam>
```
Comment (ahsonkhan) [inline xml/System.Buffers/IBufferWriter`1.xml]: This shouldn't be `ReadOnlySequence`. Where did this come from?

Comment (ahsonkhan) [reply xml/System.Buffers/IBufferWriter`1.xml]: Is there a way for me to read/review the new docs as an md/etc. and not as an xml?

The docs don't show up yet:
https://docs.microsoft.com/en-us/dotnet/api/system.buffers.ibufferwriter-1?view=netcore-2.2

Comment (mairaw) [reply xml/System.Buffers/IBufferWriter`1.xml]: The production build is still running.

You can see on review: https://review.docs.microsoft.com/en-us/dotnet/api/system.buffers.ibufferwriter-1?view=netcore-2.2&branch=master

We've made some edits on top of the /// comments. I'll change that to IBuffer.

The /// comments was a stop gap to add some docs here.

File: xml/System.Buffers/IBufferWriter`1.xml [hunk 2]
```diff
@@ -64,9 +64,9 @@
         <Parameter Name="sizeHint" Type="System.Int32" />
       </Parameters>
       <Docs>
-        <param name="sizeHint">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="sizeHint">The size of the memory span requested. If 0, requests currently available memory. </param>
+        <summary>Requests a memory span that is at least <paramref name="sizeHint" /> in size if possible.</summary>
```
Comment (ahsonkhan) [inline xml/System.Buffers/IBufferWriter`1.xml]: Should be requests a `Memory<T>`, not `memory span`. `Memory<T>` and `Span<T>` are two distinct types.

File: xml/System.Buffers/IBufferWriter`1.xml [hunk 3]
```diff
@@ -64,9 +64,9 @@
         <Parameter Name="sizeHint" Type="System.Int32" />
       </Parameters>
       <Docs>
-        <param name="sizeHint">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="sizeHint">The size of the memory span requested. If 0, requests currently available memory. </param>
```
Comment (ahsonkhan) [inline xml/System.Buffers/IBufferWriter`1.xml]: Same issue here.

File: xml/System.Buffers/BuffersExtensions.xml [hunk 4]
```diff
@@ -158,12 +158,12 @@
         <Parameter Name="value" Type="System.ReadOnlySpan&lt;T&gt;" />
       </Parameters>
       <Docs>
-        <typeparam name="T">To be added.</typeparam>
-        <param name="writer">To be added.</param>
-        <param name="value">To be added.</param>
-        <summary>To be added.</summary>
+        <typeparam name="T">The type of the items in the <see cref="T:System.Buffers.ReadOnlySequence`1" />.</typeparam>
```
Comment (ahsonkhan) [inline xml/System.Buffers/BuffersExtensions.xml]: Not readonlysequence

**Pull request merged**